src/OrleansServiceBus/Providers/Streams/EventHub/IEventHubQueueCacheFactory.cs

```diff
@@ -0,0 +1,14 @@
+﻿using Orleans.Runtime;
+using Orleans.Streams;
+
+namespace Orleans.ServiceBus.Providers
+{
+    /// <summary>
+    /// Factory responsible for creating a message cache for an EventHub partition.
+    /// </summary>
+    public interface IEventHubQueueCacheFactory
```

Comment:
Consider an Initialize call that takes the EventHubStreamProviderSettings, and SerializationManager rather than passing them through constructor.  That way the cache factory can be a configured type in the EventHubStreamProviderSettings rather than requiring one to subclass the factory and override the CreateCacheFactory.  Though the virtual function is ok, just strange having a factory factory in a factory.. feels a bit wild-westy.